lib/tests/run_tests.js

```diff
@@ -130,18 +130,20 @@ module.exports = {
           return next();
         }
 
-        runCmd(`${fs.embarkPath('node_modules/.bin/istanbul')} report --root .embark --format html`,
-          {silent: false, exitOnError: false}, (err) => {
-            if (err) {
-              return next(err);
-            }
-            console.log(`Coverage report created. You can find it here: ${fs.dappPath('coverage/index.html')}\n`);
-            const opn = require('opn');
-            const _next = () => { next(); };
-            opn(fs.dappPath('coverage/index.html'), {wait: false})
-              .then(() => new Promise(resolve => setTimeout(resolve, 1000)))
-              .then(_next, _next);
-          });
+        global.embark.events.emit('tests:finished', function() {
```

Comment:
We have `emitAndRunActionsForEvent` exactly for that

Reply:
And `registerActionForEvent`

Reply:
Maybe the controller instance in `cmd/cmd.js` should be passed to the function exported by `run_tests.js` so that it can listen on `embark.events`?

Reply:
Disregard my comments. Apparently, those functions are for plugins...

Reply:
I see what you're saying but I feel that this is a part of a bigger restructuring effort. I was surprised to see that we totally overwrite embark in this context.